spec/controllers/miq_report_controller/reports/editor_spec.rb

```diff
@@ -159,6 +159,39 @@
         end
       end
     end
+
+    describe "#categories_hash" do
+      it "returns non-empty categories" do
+        FactoryGirl.create(:classification_cost_center_with_tags)
+        FactoryGirl.create(:classification_department)
+        expect(controller.send(:categories_hash)).to eq("cc" => "Cost Center")
```

Comment:
Nice, thanks for adding tests.

Where does "cc" and "Cost Center" come from and how do I know which factory created it?  How do I know which factory should be in the `categories_hash`?

This test is hard to understand because one has to open the factories to realize the first one is creating a classification named "cc" with children, whereas the second one just creates a classification with no children.  Honestly, I think it would be beneficial to use the base classification factory and customize the parts that are important to this test.